playback/src/mixer/alsamixer.rs

```diff
@@ -0,0 +1,76 @@
+use super::AudioFilter;
+use super::{Mixer, MixerConfig};
+use std::error::Error;
+
+use alsa;
+
+#[derive(Clone)]
+pub struct AlsaMixer {
+    config: MixerConfig,
+}
+
+impl AlsaMixer {
+    fn map_volume(&self, set_volume: Option<u16>) -> Result<(u16), Box<Error>> {
+        let mixer = alsa::mixer::Mixer::new(&self.config.card, false)?;
+        let sid = alsa::mixer::SelemId::new(&*self.config.mixer, self.config.index);
+
+        let selem = mixer.find_selem(&sid).expect("Couldn't find SelemId");
+        let (min, max) = selem.get_playback_volume_range();
+        let cur_vol = selem
```

Comment:
Move this into the else statement where it's used? i.e lines 34-35

Reply:
`selem` is required to get the `min`, `max`, which is then used in both the `if` and `else` branches.
I don't think it's the neatest implementation. I have since reimplemented this by storing these `selem` specific parameters so we query it only during initialisation, and not on every volume change - but shall keep that for a separate PR.

Reply:
:+1: